doc/HorizontalReshardingGuide.md

```diff
@@ -0,0 +1,127 @@
+# Horizontal resharding guide
+
+This is a step-by-step "cookbook" style guide of instructions to follow in order to split a single-sharded keyspace into two.
```

Comment:
Other than preparing the source shard, can these instructions also be used for resharding an already multi-sharded keyspace? If so, it would be good to mention that.

Also the "step-by-step \"cookbook\" style guide" part kind of stuttered for me, because I first parsed it as "step-by-step, cookbook, (style guide)". Technically it should be "cookbook-style guide" to remove that ambiguity, but it gets awkward with the quotes. And the "guide of instructions" seems redundant. Maybe just simplify to "This is a step-by-step guide to splitting a ..."